src/analysis/retail/deathknight/unholy/modules/talents/SummonGargoyleBuffs.tsx

```diff
@@ -0,0 +1,128 @@
+import { t } from '@lingui/macro';
+import SPELLS from 'common/SPELLS';
+import TALENTS from 'common/TALENTS/deathknight';
+import { SpellLink } from 'interface';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { CastEvent } from 'parser/core/Events';
+import { When } from 'parser/core/ParseResults';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import SpellUsable from 'parser/shared/modules/SpellUsable';
+
+class SummonGargoyleBuffs extends Analyzer {
+  static dependencies = {
+    spellUsable: SpellUsable,
+  };
+
+  protected spellUsable!: SpellUsable;
+
+  totalGargoyleCasts = 0; // Number of fully completed gargoyles used
+  currentGargoyleRunicPower = 0; // Keeps track of Runic Power used on current active Gargoyle
+  totalGargoyleRunicPower = 0; // Total Runic Power spend on all fully completed Gargoyles
+  gargoyleActive = false; // Boolean to keep track of Gargoyle being active or not
+
+  constructor(options: Options) {
+    super(options);
+
+    this.active = this.selectedCombatant.hasTalent(TALENTS.SUMMON_GARGOYLE_TALENT);
+    if (!this.active) {
+      return;
+    }
+
+    this.addEventListener(
+      Events.cast
+        .by(SELECTED_PLAYER)
+        .spell([SPELLS.DEATH_COIL, SPELLS.EPIDEMIC, SPELLS.DEATH_STRIKE_HEAL]),
+      this.onBuffCast,
+    );
+
+    this.addEventListener(
+      Events.cast
+        .by(SELECTED_PLAYER)
+        .spell([SPELLS.SUMMON_GARGOYLE, SPELLS.DARK_ARBITER_TALENT_GLYPH]),
+      this.onGargCast,
+    );
+  }
+
+  onBuffCast(event: CastEvent) {
+    console.log(event.ability.name);
+    if (
+      this.gargoyleActive === true &&
+      this.spellUsable.cooldownRemaining(SPELLS.SUMMON_GARGOYLE.id, event.timestamp) < 155000
+    ) {
+      /* Gargoyle is active and cooldown is less than 2min35 seconds ->
+		A full Gargoyle has just ended. Add temporary Runic Power to total,
+		set gargoyleActive to false iterate number of Gargoyle casts by one
+		and reset the current Gargoyle Runic Power counter.*/
+      this.gargoyleActive = false;
+      this.totalGargoyleRunicPower += this.currentGargoyleRunicPower;
+      this.currentGargoyleRunicPower = 0;
+      this.totalGargoyleCasts += 1;
+    }
+    if (this.gargoyleActive === true) {
```

Comment:
Instead of two blocks checking if gargoyle is up, they should combined into a single block that checks if gargoyle is active, and then within that block we can first check if gargoyle is still up.  Then if rp should be added.

```
if (this.gargoyleActive) {

if (garg is over) {}

else if (death strike is cast){}

else (add dc/epi rp)
}

Reply:
That makes sense, will update.